examples/go/cmd/demo-audio/eventStream.go

```diff
@@ -33,36 +33,39 @@ func (manager *sessionManager) handleEvents() {
 		return
 	}
 
-	for {
-		// Wait for the next event from the server
-		event, err := eventStream.Recv()
-
-		// This indicates the stream has finished, which will happen
-		// when the session ends or the context used to create the stream
-		// closes.
-		if err == io.EOF {
-			break
+	// Start a separate go routine to handle events coming from the server
+	go func() {
+		for {
+			// Wait for the next event from the server
+			event, err := eventStream.Recv()
+
+			// This indicates the stream has finished, which will happen
+			// when the session ends or the context used to create the stream
+			// closes.
+			if err == io.EOF {
+				break
+			}
+			if err != nil {
+				textui.PrintError("Event stream error: %v\n", err)
+				return
```

Comment:
can we `break` on both or `return` on both?

Reply:
Sure